PEP 534: Improve title and clarify proposal

ncoghlan opened the pull request:
The previous title gave the impression that this PEP was about
splitting the standard library into separately versioned pieces,
when it's actually a far more modest proposal to improve the
error messages that users receive when pieces are missing, as
well as to provide a standard library API to retrieve the full
list of expected standard library module names.

ncoghlan commented:
@torsava Based on some initial feedback from @gvanrossum, I went through PEP 534 and clarified what we're actually proposing to change. Assuming yourself and @encukou think the changes look good, I'll go ahead and merge them.

torsava commented:
@ncoghlan LGTM.

As for `test.regrtest`, is there e a preference for including or not including it?

gvanrossum commented:
If you include submodules, the list will be several times longer (over 1500
instead of a few hundred). OTOH it's not inconceivable to me that some
distros would split some large packages into multiple pieces -- e.g. you
could include `test/support/*.py` but not the rest of the test package.

gvanrossum commented:
(But I don't know if distros do such things -- if we're sure they don't and
won't, the shorter list would suffice.)

encukou commented:
Fedora does include `test.support` in the main package, but not the rest of `test`. However, I consider this a [bug](https://bugzilla.redhat.com/show_bug.cgi?id=1312031), and I'd be OK with sysconfig just listing top-level modules. (That choice would also make it easier to change our minds later.)

gvanrossum commented:
OK, if Nick agrees with only returning top-level module names, then please
update the draft to clarify that. And probably add the alternative to a
"Rejected Ideas" section.

ncoghlan commented:
Aye, top-level only makes sense to me. Making that change made me realise that the PEP was missing examples for what to report when a top level package existed, but submodules were missing, so I added those.

I also made a number of other changes as the search for plausible examples brought up a few more questions of my own:

- I made it explicit that the PEP needs nominally private modules listed in the query APIs, as otherwise it won't generate the right exception messages for them
- I made the PEP more explicit about which modules it will flag as optional (and excluded the packaging modules from that set)
- I renamed the "Open Questions" section to a "Deferred Ideas" section, as the initial proposal doesn't *block* pursuing those ideas, so it makes sense to leave them out of the initial implementation

brettcannon commented:
@ncoghlan is this ready to be merged?

ncoghlan commented:
@torsava @encukou If you're happy with my updates, then we can merge this update pending @torsava getting a chance to look at what's feasible given the build system implementation in CPython.

encukou commented:
I'm happy with the updates. Thank you!

torsava commented:
LGTM, thanks Nick!
